tests/unit/test_spare_scores.py

```diff
@@ -1,14 +1,28 @@
-import sys
 import unittest
```

Comment:
Maybe time to go with pytest?

Reply:
pytest works, simply do:
``` cd tests/unit && pytest --cov=../../```

Reply:
Cool, then to make this code compatible with pytest and avoid mixing unittest syntax, could you please make the following changes:

- Remove unittest Import: We won't be using it anymore.
- Rename Test Classes: Change class names like CheckMLPDataset to TestMLPDataset. 
- Update Assertions: Replace self.assertTrue(condition) with assert condition.

Reply:
Ok, i will open a new issue for this and i will make the changes there. This can be merged if it's good